lib/rubygoal/coach.rb

```diff
@@ -16,30 +16,79 @@ def formation(match)
       raise NotImplementedError
     end
 
-    def players_errors
-      errors = {}
+    def errors
+      @errors = []
 
-      config = Rubygoal.configuration
+      check_unique_captain
+      check_players_count(:average)
+      check_players_count(:fast)
 
+      @errors
+    end
```

Comment:
I would refactor this method to avoid the need of an attribute and instead pass the errors array to the helper methods...

```ruby
def errors
  errors = []

  check_unique_captain(errors)
  check_players_count(errors, :average)
  check_players_count(errors, :fast)

  errors
end
```

It's strange how the errors attribute is being used here. Unless the idea is to cache the errors array, in that case the memoization is missing :smile: